java-checks/src/main/java/org/sonar/java/checks/LeastSpecificTypeCheck.java

```diff
@@ -96,13 +97,8 @@ private static Type findLeastSpecificType(Symbol parameter) {
         return parameter.type();
       }
       if (parent.is(Tree.Kind.FOR_EACH_STATEMENT)) {
-        // find Iterable.iterator() method
-        Collection<Symbol> symbols = parameter.type().symbol().lookupSymbols("iterator");
-        symbols.stream()
-          .filter(Symbol::isMethodSymbol)
-          .filter(s -> ((Symbol.MethodSymbol) s).parameterTypes().isEmpty())
-          .findFirst()
-          .ifPresent(inheritanceGraph::update);
+        Symbol iteratorMethod = findIteratorMethod(parameter).orElseThrow(() -> new IllegalStateException("Iterable.iterator() not found"));
```

Comment:
Please do not fail the analysis in that case : this can happen because of misconfiguration.